compiler/rustc_mir_transform/src/simplify_pow_of_two.rs

```diff
@@ -0,0 +1,212 @@
+//! A pass that checks for and simplifies calls to `pow` where the receiver is a power of
+//! two. This can be done with `<<` instead.
+
+use crate::MirPass;
+use rustc_const_eval::interpret::{ConstValue, Scalar};
+use rustc_middle::mir::patch::MirPatch;
+use rustc_middle::mir::*;
+use rustc_middle::ty::{self, Ty, TyCtxt, UintTy};
+use rustc_span::sym;
+use rustc_target::abi::FieldIdx;
+
+pub struct SimplifyPowOfTwo;
+
+impl<'tcx> MirPass<'tcx> for SimplifyPowOfTwo {
+    fn run_pass(&self, tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+        let mut patch = MirPatch::new(body);
+
+        for (i, bb) in body.basic_blocks.iter_enumerated() {
+            let term = bb.terminator();
+            let source_info = term.source_info;
+            let span = source_info.span;
+
+            if let TerminatorKind::Call {
+                    func,
+                    args,
+                    destination,
+                    target: Some(target),
+                    call_source: CallSource::Normal,
+                    ..
+                } = &term.kind
+                && let Some(def_id) = func.const_fn_def().map(|def| def.0)
+                && let def_path = tcx.def_path(def_id)
```

Comment:
Why do you need the Def path ? The crate is already 'def_id.krate'.

Reply:
Artifact from how it worked before